### PR TITLE
Adjust CSV generation for OCP platform

### PR DESCRIPTION
**Pull request by salv-orlando**
Modify RBAC properties to ensure appropriate deployment via OLM.
Also ensure deploy/role.yaml has the same contents as those
specified in kubebuilder annotations for antreainstall_controller.go.

This commit also adds a dedicated Makefile to build a bundle for
the Openshift platform which adds some specific annotations.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -128,10 +128,14 @@ endif
 .PHONY: bundle
 bundle: manifests kustomize
 	operator-sdk generate kustomize manifests -q
-	cd config/manager && $(KUSTOMIZE) edit set image antrea-operator=antrea/antrea-operator:$(VERSION)
-	$(KUSTOMIZE) build config/manifests | operator-sdk generate bundle -q --overwrite --version $(VERSION) $(BUNDLE_METADATA_OPTS)
```
Comment (salv-orlando) [inline Makefile:132]: The --version attribute was not necessary, hence removing it

File: bundle.Dockerfile [hunk 2]
```diff
@@ -5,11 +5,15 @@ LABEL operators.operatorframework.io.bundle.manifests.v1=manifests/
...
 LABEL operators.operatorframework.io.metrics.mediatype.v1=metrics+v1
 LABEL operators.operatorframework.io.metrics.project_layout=go.kubebuilder.io/v2
 LABEL operators.operatorframework.io.test.config.v1=tests/scorecard/
 LABEL operators.operatorframework.io.test.mediatype.v1=scorecard+v1
+
 COPY bundle/manifests /manifests/
 COPY bundle/metadata /metadata/
 COPY bundle/tests/scorecard /tests/scorecard/
+LABEL com.redhat.openshift.versions=v4.6,v4.7
+LABEL com.redhat.delivery.operator.bundle=true
```
Comment (salv-orlando) [inline bundle.Dockerfile:19]: The additional labels are required for Openshift certification

File: bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml [hunk 3]
```diff
@@ -1,8 +1,8 @@
-apiVersion: apiextensions.k8s.io/v1beta1
+apiVersion: apiextensions.k8s.io/v1
```
Comment (salv-orlando) [inline bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml:1]: The change  in this CR is due to the update from v1beta1 to v1

File: controllers/antreainstall_controller.go [hunk 4]
```diff
@@ -63,20 +63,25 @@ func (r *AntreaInstallReconciler) SetupWithManager(mgr ctrl.Manager) error {
...
+// +kubebuilder:rbac:groups=authentication.k8s.io,resources=tokenreviews,verbs=create
 // +kubebuilder:rbac:groups=networking.k8s.io,resources=networkpolicies,verbs=get;watch;list
 // +kubebuilder:rbac:groups=ops.antrea.tanzu.vmware.com,resources=traceflows;traceflows/status,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=clusterinformation.antrea.tanzu.vmware.com,resources=antreaagentinfos;antreacontrollerinfos,verbs=get;list;create;update;delete
 // +kubebuilder:rbac:groups=networking.antrea.tanzu.vmware.com,resources=networkpolicies;appliedtogroups;addressgroups,verbs=get;watch;list;delete
 // +kubebuilder:rbac:groups=security.antrea.tanzu.vmware.com,resources=clusternetworkpolicies,verbs=get;watch;list;delete
 // +kubebuilder:rbac:groups=system.antrea.tanzu.vmware.com,resources=controllerinfos;agentinfos;supportbundles;supportbundles/download,verbs=get;watch;list;post;delete
 // +kubebuilder:rbac:urls=/agentinfo;/addressgroups;/appliedtogroups;/networkpolicies;/ovsflows;/ovstracing;/podinterfaces,verbs=get
+// +kubebuilder:rbac:groups=security.openshift.io,resources=securitycontextconstraints,resourceNames=hostnetwork,verbs=use
+
```
Comment (salv-orlando) [inline controllers/antreainstall_controller.go:84]: As a follow up we should probably find a way to generate deploy/role.yaml from these annotations as well

File: bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml (hunk 3 above)
Comment (jwsui) [reply bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml:1]: Should the `deploy/operator.antrea.vmware.com_antreainstalls_crd.yaml` be updated to be consistent with this file?

Comment (jwsui) [inline bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml:1]: Should the `deploy/operator.antrea.vmware.com_antreainstalls_crd.yaml` be updated to be consistent with this file?

Comment (salv-orlando) [reply bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml:1]: It's a good idea indeed. I'l take care of it.

Comment (salv-orlando) [reply bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml:1]: @jwsui I noticed the operator repo has both the v1beta1 and v1 CR in https://github.com/vmware/antrea-operator-for-kubernetes/blob/main/deploy/. I think we might want to leave this PR as it is and push a new PR to remove the v1beta1 version for the CR.

Comment (salv-orlando) [reply bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml:1]: New Pr: https://github.com/vmware/antrea-operator-for-kubernetes/pull/29

Comment (jwsui) [reply bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml:1]: ok, removing redundant crd makes things easier.

**Review by jwsui — APPROVED** [review]

**Pull request merged**